pandas/tests/groupby/test_pipe.py

```diff
@@ -55,25 +55,22 @@ def test_pipe_args():
     )
 
     def f(dfgb, arg1):
-        return dfgb.filter(lambda grp: grp.y.mean() > arg1, dropna=False).groupby(
-            dfgb.grouper
-        )
+        filtered = dfgb.filter(lambda grp: grp.y.mean() > arg1, dropna=False)
+        return filtered.groupby("group")
```

Comment:
just for my understanding, why did this need to change?

Reply:
When using `.groupby(dfgb.grouper)`, the column `group` in included in the operation. While the `sum` is successful, the function `g` then adds `arg2` (with value 10) and this fails. We could also supply `numeric_only=True` to the two `sum` operations in `g`, but this seems a little easier.